plugins/modules/docker_image.py

```diff
@@ -879,6 +897,80 @@ def load_image(self):
             return self.client.find_image(self.name, self.tag)
 
 
+class ImageArchiveInvalidException(Exception):
+    pass
+
+
+def archived_image_id(archive_path):
+    """
+    Attempts to get Image.Id from meta data stored in the image archive tar file.
+
+    The tar should contain a file "manifest.json" with an array with a single entry,
+    and the entry should have a Config field with the image ID in its file name.
+
+    Raises:
+        ImageArchiveInvalidException if a file already exists at archive_path, but
+        we could not extract an image ID from it.
+
+    Returns:
+        Either None, if no file exists at archive_path, or the extracted image ID.
+        The extracted ID will not have a sha256: prefix.
+
+    :return str
+    """
+
+    try:
+        # FileNotFoundError does not exist in Python 2
+        if not os.path.isfile(archive_path):
+            return None
+
+        tf = tarfile.open(archive_path, 'r')
+        try:
+            try:
+                ef = tf.extractfile('manifest.json')
+
+                if hasattr(ef, '__exit__'):
+                    # Python 3
```

Comment:
Better use `PY2` (`from ansible.module_utils.six import PY2`) to handle these differences.